riscv-sbi.adoc

```diff
@@ -1674,6 +1675,250 @@ The possible error codes returned in `sbiret.error` are shown in the
+As an example, it could be a cryptographic nonce that must be bound to the
+generated evidence in order to protect the remote attestation process from
+replay attacks.
+
+The SBI implemenation replies to the signing request by generating a
+https://datatracker.ietf.org/doc/html/rfc5280[X.509] formatted certificate at
+the `cert_addr` physical address. The generated certificate must be signed with
+a platform attestation keys and must include a hash of the data blob pointed by
+`report_data_addr`. Its actual length is in `sbiret.value` and must not exceed
+`max_cert_size` bytes
+
+Moreover the generated certificate must contain:
+
+* A X.509 extension formatted according to `evidence_format`, containing the
+attestation evidence that is associated with the CSR `Subject` key. In other
+words, the generated certificate `Subject` and `SubjectPublicKey` identify the
+entity to which this extension defined evidence applies to.
+`evidence_format` must be one of the supported evidence formats, as reported by
+the `EVIDENCE_FORMATS` field from the <<get-capabilities-function>> returned
+structure (`ATST_CAPS`). See
+<<table_attestation_and_measurement_capabilities_format>> for more details.
+* A https://oidref.com/2.5.29.49[UserNotice] X.509 extension containing the
+the data blob from `evidence_data_addr`.
+* A https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1[AuthorityKeyIdentifier]
+X.509 extension.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_get_evidence_errors>> below.
+
+[#table_get_evidence_errors]
+.Attestation Get Evidence Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code          | Description
+| SBI_SUCCESS         | The evidence is generated and the certificate size is
+                        `sbiret.value`.
+| SBI_ERR_FAILED      | The evidence could not be generated.
+| SBI_INVALID_ADDRESS | The CSR or the certificate adress is invalid.
+| SBI_INVALID_PARAM   | The CSR at `csr_addr` is invalid.
+|===
+
+[NOTE]
+.Attestation Flow Example
+====
+Guest userspace flow TBD
+====
+
+[#measurement-extend-function,reftext="Extend Measurement"]
+=== Function: Extend Measurement (FID #1)
+
+[source, C]
+----
+struct sbiret sbi_measurement_extend(unsigned long measurement_addr,
+                                     unsigned long measurement_index,
+                                     unsigned long hash_algorithm)
+----
+
+Extend the platform TCB measurement with one additional measurement.
+
+This function typically allows the caller to add a software component to the
+list of the TCB layers measurements at any point of the platform lifecycle.
+By supporting this function, the SBI implementation implicitly becomes a
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[Dynamic Root of Trust for Measurement (DRTM)].
+
+Measurements added through this SBI function should be managed separately from
```

Comment:
Clarify actor through use of active voice: "The SBI implementation should manage measurement added through this SBI function separately. It can choose to..."